### PR TITLE
zh_CN:  translation  js_guide_save_load.md

### PR DESCRIPTION
**Pull request by PeterJaq**

### CODE REVIEW

**Comment by lamberta** [comment]
Thanks, Jiang.

@duanw0916, can you please find a reviewer for this pull request? Thank you

File: site/zh-cn/js/guide/save_load.md [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+**Scheme:** `downloads://`
+
+```js
+await model.save('downloads://my-model');
+```
+这会让浏览器下载模型文件至用户的机器上，并生成两个文件：
+ 1. 一个名为 `[my-model].json` 的 JSON 文件，它在 modeloTopology 字段中包含模型的拓扑，在 weightsManifest 字段中包含权重清单。
+ 2. 一个二进制文件，其中包含名为 `[my-model].weights.bin` 的权重值。
+
+你可以更换`[my-model]`的名称以获得一个不同的文件名。
```
Comment (kjlintong) [inline site/zh-cn/js/guide/save_load.md]:
```suggestion
你可以更换`[my-model]`的名称以获得一个不同的名称的文件。
```

File: site/zh-cn/js/guide/save_load.md [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+**Scheme:** `http://` or `https://`
+
+```js
+await model.save('http://model-server.domain/upload')
+```
+
+这里将会创建一个web请求以保存一个模型在远程服务器上。您应该控制该远程服务器，以便确保它能够处理请求。
+则模型的拓扑和权重将通过[POST](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/POST) 请求发送到指定的 HTTP 服务器。
+POST 请求的 body 具有称为`multipart/form-data`的格式。
```
Comment (kjlintong) [inline site/zh-cn/js/guide/save_load.md]:
```suggestion
POST 请求的 body 遵守称为`multipart/form-data`的格式。它由以下两个文件组成
```

File: site/zh-cn/js/guide/save_load.md [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+
+ 1. 一个名为 `model.json` 的 JSON 文件，其中包含拓扑结构和对下面描述的权重文件的引用。
+2.  一个二进制文件，其中包含名为`model.weights.bin`. 的权重值。
+
+这个 [api 文档](https://js.tensorflow.org/api/latest/#tf.io.browserHTTPRequest)包含一块 Python 代码片段，演示了如何使用加入了 Keras 和 TensorFlow 的 [flask](http://flask.pocoo.org/) Web 框架来处理来自 save 的请求，并重建 request 的 payload 作为服务器内存中的 Keras Model 对象。
+
+通常，您的 HTTP 服务器对请求有特殊约束和要求，例如 HTTP 中的 methods（方法）、headers（头部）和 credentials for authentication（身份验证凭据）。 您可以通过替换 `tf.io.browserHTTPRequest` 调用的 URL 字符串参数来获得 save 函数在上述方面更细粒度的控制。 它是一个更详细的 API ，但它在控制由 save 发起的 HTTP 请求方面提供了更大的灵活性。
```
Comment (kjlintong) [inline site/zh-cn/js/guide/save_load.md]:
```suggestion
通常，您必须向HTTP服务器传递更多参数或请求标头（例如，用于身份验证，或者如果要指定应保存模型的文件夹）。您可以通过替换URL字符串参数tf.io.browserHTTPRequest来获得对来自 save 函数的请求的这些方面的细粒度控制。此API在控制HTTP请求方面提供了更大的灵活性。
```

File: site/zh-cn/js/guide/save_load.md [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+- 类似于`model.save()`,  `loadLayersModel`函数使用以 **scheme**开头的类似URL的字符串参数。它描述了我们试图从中加载模型的目标类型。
+- scheme 由**path**指定。在上述例子中路径为`my-model-1`。
+- URL字符串可以被替换为一个符合IOHandler接口的对象。
+- `tf.loadLayersModel()`函数是异步的。
+- `tf.loadLayersModel`返回的值是 `tf.Model`
+
+下面我们将介绍可用的不同方案。
+
+
+### Local Storage (Browser only)
```
Comment (kjlintong) [inline site/zh-cn/js/guide/save_load.md]:
```suggestion
### 本地存储 (仅限浏览器)
```

File: site/zh-cn/js/guide/save_load.md [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+**Scheme:** `localstorage://`
+
+```js
+const model = await tf.loadLayersModel('localstorage://my-model');
+```
+
+这将从浏览器的[本地存储](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage).
+加载一个名为`my-model`模型。
+
+### IndexedDB (Browser only)
```
Comment (kjlintong) [inline site/zh-cn/js/guide/save_load.md]:
```suggestion
### IndexedDB (仅限浏览器)
```

**Comment by kuri-leo** [comment]
> Thanks, Jiang.
> 
> @duanw0916, can you please find a reviewer for this pull request? Thank you

Hi @lamberta, Sorry for my temporary disappearance but it will continue to the next week as I'm deeply trapped in the nightmare of final exams 🤦‍♂ 

I will be back as soon as my college troubles were finished (or I was finished, and review those left PRs & give patches to the merged (They seem to have problems left ).

Thanks

**Comment by lamberta** [comment]
@kuri-leo No worries at all! Appreciate all your work here to build the community and please do well on your exams :)

Some more reviewers have graciously volunteered to help. FYI @JayYip, @yantaozhao, @loveunk, @Wind2esg

Thank you, all!

**Comment by kuri-leo** [comment]
mark, review during weekend

File: site/zh-cn/js/guide/save_load.md [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+```
+
+这里将会创建一个web请求以保存一个模型在远程服务器上。您应该控制该远程服务器，以便确保它能够处理请求。
+则模型的拓扑和权重将通过[POST](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/POST) 请求发送到指定的 HTTP 服务器。
+POST 请求的 body 遵守称为`multipart/form-data`的格式。它由以下两个文件组成
+
+ 1. 一个名为 `model.json` 的 JSON 文件，其中包含拓扑结构和对下面描述的权重文件的引用。
+2.  一个二进制文件，其中包含名为`model.weights.bin`. 的权重值。
+
+这个 [api 文档](https://js.tensorflow.org/api/latest/#tf.io.browserHTTPRequest)包含一块 Python 代码片段，演示了如何使用加入了 Keras 和 TensorFlow 的 [flask](http://flask.pocoo.org/) Web 框架来处理来自 save 的请求，并重建 request 的 payload 作为服务器内存中的 Keras Model 对象。
```
Comment (kuri-leo) [inline site/zh-cn/js/guide/save_load.md]:
```suggestion
请注意，这两个文件的名称需要与上述介绍中的保持完全相同（因为名称内置于函数中，无法更改）。 此[ api 文档](https://js.tensorflow.org/api/latest/#tf.io.browserHTTPRequest)包含一个 Python 代码片段，演示了如何使用 [flask](http://flask.pocoo.org/) web 框架来处理源自 `save` 的请求。
```

File: site/zh-cn/js/guide/save_load.md [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+
+通常，您必须向HTTP服务器传递更多参数或请求标头（例如，用于身份验证，或者如果要指定应保存模型的文件夹）。您可以通过替换URL字符串参数tf.io.browserHTTPRequest来获得对来自 save 函数的请求的这些方面的细粒度控制。此API在控制HTTP请求方面提供了更大的灵活性。
```
Comment (kuri-leo) [inline site/zh-cn/js/guide/save_load.md]:
```suggestion
通常，您必须向 HTTP 服务器传递更多参数或请求头（例如，用于身份验证，或者如果要指定应保存模型的文件夹）。您可以通过替换 `tf.io.browserHTTPRequest` 函数中的 URL字符串参数来获得对来自 `save` 函数的请求在这些方面的细粒度控制。这个API在控制 HTTP 请求方面提供了更大的灵活性。
```

File: site/zh-cn/js/guide/save_load.md [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+
+ 1. 一个名为 `model.json` 的 JSON 文件，其中包含拓扑结构和对下面描述的权重文件的引用。
+1.  一个二进制文件，其中包含名为`model.weights.bin`. 的权重值。
+
+请注意，这两个文件的名称将始终与上面指定的完全相同（该名称内置于函数中）。
+
+
+## 加载 tf.Model
+
+给定一个使用上述方法之一保存的模型，我们可以使用 `tf.loadLayersModel`api加载它。
```
Comment (kuri-leo) [inline site/zh-cn/js/guide/save_load.md]:
```suggestion
给定一个使用上述方法之一保存的模型，我们可以使用 `tf.loadLayersModel` API来加载它。
```

File: site/zh-cn/js/guide/save_load.md [hunk 9]
```diff
@@ -0,0 +1,202 @@
...
+### HTTP(S)
+
+**Scheme:** `http://` or `https://`
+
+```js
+const model = await tf.loadLayersModel('http://model-server.domain/download/model.json');
+```
+这将从HTTP端点加载模型。加载`json` 文件后，函数将请求对应的`json` 文件引用的`.bin`文件。
+
+> 注意：这个工具依赖于[`fetch`](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch)方法。如果你的环境没有提供原生的fetch方法，你可以提供全局方法名称[`fetch`](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch)从而满足接口要求或是使用类似于(`node-fetch`)[https://www.npmjs.com/package/node-fetch]的库。
```
Comment (kuri-leo) [inline site/zh-cn/js/guide/save_load.md]:
```suggestion
> 注意：这个工具依赖于[`fetch`](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch)方法。如果您的环境没有提供原生的fetch方法，您可以提供全局方法名称[`fetch`](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch)从而满足接口要求或是使用类似于(`node-fetch`)[https://www.npmjs.com/package/node-fetch]的库。
```

File: site/zh-cn/js/guide/save_load.md [hunk 10]
```diff
@@ -0,0 +1,202 @@
...
+
+```js
+const model = await tf.loadLayersModel('file://path/to/my-model/model.json');
+```
+
+当运行在Node.js上时，我们可以直接访问文件系统并且从那里加载模型。注意，在上面的函数调用中，我们引用model.json文件本身（而在保存时，我们指定一个文件夹）。相应的`.bin`文件需要和`json` 文件在同一个文件夹中。
+
+## 使用 IOHandlers 加载模型
+
+如果上述方案没有满足你的需求，您可以使用`IOHandler`执行客制化加载行为。Tensorflow.js的`IOHandler`提供了[`tf.io.browserFiles`](https://js.tensorflow.org/api/latest/#io.browserFiles) ，运行浏览器用户在浏览器中上传文件。您可以在 [文档](https://js.tensorflow.org/api/latest/#io.browserFiles)中查看更多信息。
```
Comment (kuri-leo) [inline site/zh-cn/js/guide/save_load.md]:
```suggestion
如果上述方案没有满足您的需求，您可以使用`IOHandler`执行自定义的加载行为。Tensorflow.js的`IOHandler`提供了[`tf.io.browserFiles`](https://js.tensorflow.org/api/latest/#io.browserFiles) ，运行浏览器用户在浏览器中上传文件。您可以在 [文档](https://js.tensorflow.org/api/latest/#io.browserFiles)中查看更多信息。
```

File: site/zh-cn/js/guide/save_load.md [hunk 11]
```diff
@@ -0,0 +1,202 @@
...
+
+# 保存和加载模型使用客制化IOHandlers
```
Comment (kuri-leo) [inline site/zh-cn/js/guide/save_load.md]:
```suggestion
# 使用自定义的 IOHandlers 保存或加载模型
```

File: site/zh-cn/js/guide/save_load.md [hunk 12]
```diff
@@ -0,0 +1,202 @@
...
+
+如果上述方案没有满足你的保存和加载模型的需求，你可以通过执行`IOHandler`以执行客制化序列行为。
```
Comment (kuri-leo) [inline site/zh-cn/js/guide/save_load.md]:
```suggestion
如果上述方案没有满足您的保存和加载模型的需求，您可以通过执行`IOHandler`以执行自定义的序列化行为。
```

**Pull request merged**